js-pkg/client/src/provider.ts

```diff
@@ -182,22 +264,21 @@ export class YSweetProvider {
       syncProtocol.writeUpdate(encoder, update)
       this.send(encoding.toUint8Array(encoder))
 
+      this.incrementLocalVersion()
       this.checkSync()
     }
   }
 
   private checkSync() {
-    this.lastSyncSent += 1
     const encoder = encoding.createEncoder()
     encoding.writeVarUint(encoder, MESSAGE_SYNC_STATUS)
 
     const versionEncoder = encoding.createEncoder()
-    encoding.writeVarUint(versionEncoder, this.lastSyncSent)
+    encoding.writeVarUint(versionEncoder, this.localVersion)
     encoding.writeVarUint8Array(encoder, encoding.toUint8Array(versionEncoder))
 
     this.send(encoding.toUint8Array(encoder))
-
-    this.updateSyncedState()
```

Comment:
Just making sure this is intentional — we do _not_ want to emit the `local-changes` event here?

Reply:
that's right, this function is now responsible for checking if the _current_ `localVersion` has been seen by the server. Since it is no longer responsible for incrementing `localVersion`, it can no longer lead to the local sync status changing -- it is the responsibility of the caller to increment `localVersion` and emit the event if necessary.